### PR TITLE
feat(api): enhance Actions Secrets Management API for repository

### PR DESCRIPTION
**Pull request by appleboy**
- Add endpoint to list repository action secrets in API routes
- Implement `ListActionsSecrets` function to retrieve action secrets from the database
- Update Swagger documentation to include the new `/repos/{owner}/{repo}/actions/secrets` endpoint
- Add `actions` package import and define new routes for actions, secrets, variables, and runners in `api.go`.
- Refactor action-related API functions into `Action` struct methods in `org/action.go` and `repo/action.go`.
- Remove `actionAPI` struct and related functions, replacing them with `NewAction()` calls.
- Rename `variables.go` to `action.go` in `org` directory.
- Delete `runners.go` and `secrets.go` in both `org` and `repo` directories, consolidating their content into `action.go`.
- Update copyright year and add new imports in `org/action.go`.
- Implement `API` interface in `services/actions/interface.go` for action-related methods.
- Remove individual action-related functions and replace them with methods on the `Action` struct in `repo/action.go`.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
It's better to have integration tests.

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -1596,6 +1585,47 @@ func Routes() *web.Route {
...
+	ListActionsSecrets   func(ctx *context.APIContext)
+	CreateOrUpdateSecret func(ctx *context.APIContext)
+	DeleteSecret         func(ctx *context.APIContext)
+	ListVariables        func(ctx *context.APIContext)
+	GetVariable          func(ctx *context.APIContext)
+	DeleteVariable       func(ctx *context.APIContext)
+	CreateVariable       func(ctx *context.APIContext)
+	UpdateVariable       func(ctx *context.APIContext)
+	GetRegistrationToken func(ctx *context.APIContext)
+}
```
Comment (wxiaoguang) [inline routers/api/v1/api.go]: This struct looks quite strange.

I think there are some samples for such usage, eg: 

https://github.com/go-gitea/gitea/blob/dd2aaadce3ecd3134a1ba0c82c5aaa05d6c11b2b/routers/web/web.go#L442-L448

Comment (appleboy) [reply routers/api/v1/api.go:1599]: I will try it.

Comment (appleboy) [reply routers/api/v1/api.go]: @wxiaoguang lint error: lines are duplicate

Maybe we can change the action API struct to interface?

Comment (wxiaoguang) [reply routers/api/v1/api.go]: No idea why "lines are duplicate" would happen. Such code should indeed avoid duplication.

Comment (appleboy) [reply routers/api/v1/api.go]: image

image

@wxiaoguang  See.

Comment (wxiaoguang) [reply routers/api/v1/api.go]: Since don't duplicate them, only use one common function?

----

Oops ..... maybe the different package name is the cause .....

File: routers/api/v1/api.go [hunk 2]
```diff
@@ -1460,27 +1455,21 @@ func Routes() *web.Route {
...
+					org.ListActionsSecrets,
+					org.CreateOrUpdateSecret,
+					org.DeleteSecret,
+					org.ListVariables,
+					org.GetVariable,
+					org.DeleteVariable,
+					org.CreateVariable,
+					org.UpdateVariable,
+					org.GetRegistrationToken,
+				},
```
Comment (wxiaoguang) [inline routers/api/v1/api.go]: I think it should use the field names explicitly. Otherwise, it's difficult to know whether the function really matches the field name: eg: `Struct{Func1, Func2, Func3, Func4}` and `Struct{Func1, Func3, Func2, Func4}` all could compile and run, but mismatched route handlers would cause problems.

File: routers/api/v1/org/action.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2024 The Gitea Authors. All rights reserved.
+// Copyright 2023 The Gitea Authors. All rights reserved.
```
Comment (lunny) [inline routers/api/v1/org/action.go]: Unnecessary change.

**Review by wxiaoguang — APPROVED** [review]
Overall LGTM

File: routers/api/v1/org/action.go (hunk 3 above)
Comment (appleboy) [reply routers/api/v1/org/action.go]: https://github.com/go-gitea/gitea/pull/30656/commits/2f94488c5e7ad98156d03fb4feefa3e6e64df7dd

File: tests/integration/api_repo_secrets_test.go [hunk 4]
```diff
@@ -24,14 +24,20 @@ func TestAPIRepoSecrets(t *testing.T) {
...
+
 	t.Run("Create", func(t *testing.T) {
 		cases := []struct {
 			Name           string
 			ExpectedStatus int
 		}{
 			{
 				Name:           "",
-				ExpectedStatus: http.StatusNotFound,
+				ExpectedStatus: http.StatusMethodNotAllowed,
```
Comment (lunny) [inline tests/integration/api_repo_secrets_test.go:40]: Why did the result change?

File: routers/api/v1/api.go [hunk 5]
```diff
@@ -835,6 +836,34 @@ func Routes() *web.Route {
...
 	}))
 
+	addActionsRoutes := func(
+		m *web.Route,
+		reqChecker func(ctx *context.APIContext),
+		act actions.API,
+	) {
+		m.Group("/actions", func() {
+			m.Group("/secrets", func() {
+				m.Get("", reqToken(), reqChecker, act.ListActionsSecrets)
```
Comment (appleboy) [inline routers/api/v1/api.go:846]: I add a new route for list action secrets. @lunny

File: tests/integration/api_repo_secrets_test.go (hunk 4 above)
Comment (appleboy) [reply tests/integration/api_repo_secrets_test.go:40]: See the comment https://github.com/go-gitea/gitea/pull/30656/files#r1579573821

File: routers/api/v1/api.go [hunk 6]
```diff
@@ -1073,26 +1102,11 @@ func Routes() *web.Route {
 					m.Post("/accept", repo.AcceptTransfer)
 					m.Post("/reject", repo.RejectTransfer)
 				}, reqToken())
-				m.Group("/actions", func() {
-					m.Group("/secrets", func() {
```
Comment (appleboy) [inline routers/api/v1/api.go:1077]: The origin code of repository route doesn't support list secrets. @lunny

**Review by lunny — APPROVED** [review]
